.github/workflows/ci.yml

```diff
@@ -13,57 +13,58 @@ on:
 jobs:
 
   build:
-    name: Python==${{ matrix.python-version }} | ${{ matrix.django-version }}
-    runs-on: ubuntu-20.04
+    name: Python==${{ matrix.python-version }}
+    runs-on: ubuntu-22.04
     strategy:
       fail-fast: false
       matrix:
         python-version:
-          - 3.7
-          - 3.8
-          - 3.9
+          - "3.8"
+          - "3.9"
+          - "3.10"
 
     steps:
     - uses: actions/checkout@v2
       with:
         ref: ${{ github.event.pull_request.head.sha }}
 
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+        cache-dependency-path: |
+          **/requirements*.txt
 
-    - name: Upgrade python system packages
-      run: pip install -U pip wheel
-
-    - name: Install test dependencies
-      run: pip install -r requirements-test.txt
-
-    - name: Install netdiff
-      run: pip install -e .
+    - name: Install Dependencies
+      id: deps
+      run: |
+        pip install -U pip wheel setuptools
+        pip install -U -r requirements-test.txt
+        pip install -U -e .
+        pip install --no-cache-dir -U "openwisp-utils[qa] @ https://github.com/openwisp/openwisp-utils/tarball/master"
```

Comment:
This should go here: https://github.com/openwisp/netdiff/blob/master/requirements-test.txt